### DISCUSSION TITLE
Git workflow for deployment

### DISCUSSION THREAD

**Discussion by vanyell** [horilla-opensource/horilla#362] [General]
#### Basically, I did the following steps:
*Initial*
1.  git clone 
2.  git checkout  -b dev
3. make change in dev branch

*succeeding*
1. git checkout master 
2. git pull && git rebase
3. git checkout dev && git merge master
4. source venv/bin/activate
5. python manage.py makemigrations
6. python manage.py migrate

*Pushing to web directory*
1. git push /var/www/html/horilla
2. python manage.py makemigrations
3. python manage.py migrate

Since venv is ignored. I have to instantiate a separate venv in the web directory

Any comments on these steps?

**Comment by horilla-opensource** [comment]
1. **Set Up the Local Repository with Git**
    If the user has already cloned the Horilla repository, navigate to the local folder:
   `cd path/to/horilla`

    If not, clone it from the GitHub repository:
   ```
    git clone https://github.com/horilla-opensource/horilla.git
    cd horilla
   ```
2. **Stash Local Changes (If Any)**
    If the user has made local changes to the code (e.g., minor customizations), they should 
    stash or commit them to prevent merge conflicts. Use:
    `git stash`

    Or commit the changes:

    ```
    git add .
    git commit -m "Local custom changes"
    ```

3. **Pull the Latest Changes from the master Branch**
    To keep the local repository updated with the latest changes from the master branch:
    `git pull origin master`
    This command will fetch the latest changes from the remote repository and merge them into 
    the local master branch. Since the database is not modified by this operation, the user will 
    not lose any data.

4. **Check for Any Required Database Migrations**
    Even though you mentioned that the database remains unchanged, sometimes code updates 
    include new database migrations. To ensure that the codebase and database schema are 
    compatible, run:
    ```
    python manage.py makemigrations
    python manage.py migrate
    ```
5. Restart the Application
    `python manage.py runserver`

**Reply by JVKeller** [reply]
If the server is already running, is there a command to stop/restart it?

```
root@1d386e6368c5:/app# python manage.py runserver
Watching for file changes with StatReloader
Performing system checks...

System check identified no issues (0 silenced).
Error: That port is already in use.
root@1d386e6368c5:/app# 
```

**Reply by horilla-opensource** [reply]
Hi @JVKeller ,

To have a better workflow management, you can create a service file for the application and run the server using the service file.

Please find a detailed documentation on how to deploy Horilla using nginx and gunicorn.

[Horilla Deployment Guide: Using Gunicorn and Nginx](https://docs.google.com/document/d/15XQmAqSPUM5i8ORWYdZSMm-zBEabAbS0JGH9VLIjm_Y/edit?usp=sharing)
